jax_galsim/moffat.py

```diff
@@ -0,0 +1,467 @@
+        x1 = self.flux * 4
+        x2 = jnp.power(2.0, self.beta)
+        x3 = jnp.exp(jax.lax.lgamma(self.beta - 1.0))
+        return x1 / (x2 * x3)
+
+    def __hash__(self):
+        return hash(
+            (
+                "galsim.Moffat",
+                self.beta,
+                self.scale_radius,
+                self.trunc,
+                self.flux,
+                self.gsparams,
+            )
+        )
+
+    def __repr__(self):
+        return (
+            "galsim.Moffat(beta=%r, scale_radius=%r, trunc=%r, flux=%r, gsparams=%r)"
+            % (self.beta, self.scale_radius, self.trunc, self.flux, self.gsparams)
+        )
+
+    def __str__(self):
+        s = "galsim.Moffat(beta=%s, scale_radius=%s" % (self.beta, self.scale_radius)
+        if self.trunc != 0.0:
+            s += ", trunc=%s" % self.trunc
+        if self.flux != 1.0:
+            s += ", flux=%s" % self.flux
+        s += ")"
+        return s
+
+    @property
+    def _maxk_untrunc(self):
+        """untruncated Moffat maxK
+
+            The 2D Fourier Transform of f(r)=C (1+(r/rd)^2)^(-beta) leads
+             C rd^2 = Flux (beta-1)/pi (no truc)
+             and
+             f(k) = C rd^2 int_0^infty (1+x^2)^(-beta) J_0(krd x) x dx
+                  = 2 F (k rd /2)^(\beta-1) K[beta-1, k rd]/Gamma[beta-1]
+            with k->\infty asymptotic behavior
+                 f(k)/F \approx sqrt(pi)/Gamma(beta-1) e^(-k') (k'/2)^(beta -3/2) with k' = k rd
+            So we solve f(maxk)/F = thr  (aka maxk_threshold  in  gsparams.py)
+                leading to the iterative search of
+                let alpha = -log(thr Gamma(beta-1)/sqrt(pi))
+                k = (\beta -3/2)\log(k/2) + alpha
+                starting with k = alpha
+
+        note : in the code "alternative code" is related to issue #1208 in GalSim github
+        """
+
+        def body(i, val):
+            kcur, alpha = val
+            knew = (self.beta - 0.5) * jnp.log(kcur) + alpha
+            ## knew = (self.beta -1.5)* jnp.log(kcur/2) + alpha # alternative code
+            return knew, alpha
+
+        ## alpha = -jnp.log(self.gsparams.maxk_threshold * jnp.exp(jsc.special.gammaln(self._beta-1))/jnp.sqrt(jnp.pi) ) # alternative code
+
+        alpha = -jnp.log(
+            self.gsparams.maxk_threshold
+            * jnp.power(2.0, self.beta - 0.5)
+            * jnp.exp(jsc.special.gammaln(self.beta - 1))
+            / (2 * jnp.sqrt(jnp.pi))
+        )
+
+        val_init = (
+            alpha,
+            alpha,
+        )
+        val = jax.lax.fori_loop(0, 5, body, val_init)
+        maxk, alpha = val
+        return maxk / self._r0
+
+    @property
+    def _prefactor(self):
+        return 2.0 * (self.beta - 1.0) / (self._fluxFactor)
+
+    @property
+    def _maxk_trunc(self):
+        """truncated Moffat maxK"""
+        maxk_val = (
+            self.gsparams.maxk_threshold
+        )  # a for gaussian profile... this is f(k_max)/Flux = maxk_threshold
```

Comment:
```suggestion
        # a for gaussian profile... this is f(k_max)/Flux = maxk_threshold
        maxk_val = self.gsparams.maxk_threshold
```

Reply:
try to avoid multi-line commands, makes it harder to read

Reply:
done